src/main/java/com/magicalteam/authentication/AttestationDecoder.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The contents of this file are subject to the terms of the Common Development and
+ * Distribution License (the License). You may not use this file except in compliance with the
+ * License.
+ *
+ * You can obtain a copy of the License at legal/CDDLv1.0.txt. See the License for the
+ * specific language governing permission and limitations under the License.
+ *
+ * When distributing Covered Software, include this CDDL Header Notice in each file and include
+ * the License file at legal/CDDLv1.0.txt. If applicable, add the following below the CDDL
+ * Header, with the fields enclosed by brackets [] replaced by your own identifying
+ * information: "Portions copyright [year] [name of copyright owner]".
+ *
+ * Copyright 2018 ForgeRock AS.
+ */
+package com.magicalteam.authentication;
+
+import java.io.ByteArrayInputStream;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.List;
+
+import co.nstant.in.cbor.CborDecoder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.Number;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.Map;
+import co.nstant.in.cbor.model.UnicodeString;
+
+class AttestationDecoder {
+
+    FidoAttestationObject decode(byte[] attestationData) {
+
+
+        FidoAttestationObject attestationObject = new FidoAttestationObject();
+        ByteArrayInputStream bais = new ByteArrayInputStream(attestationData);
+        List<DataItem> dataItems = null;
+        try {
+            dataItems = new CborDecoder(bais).decode();
+        } catch (CborException e) {
+            e.printStackTrace();
+        }
+        Map attObjMap = (Map) dataItems.get(0);
+        for (DataItem key : attObjMap.getKeys()) {
+            if (key instanceof UnicodeString) {
+                if (((UnicodeString) key).getString().equals("fmt")) {
+                    UnicodeString value = (UnicodeString) attObjMap.get(key);
+                    attestationObject.fmt = value.getString();
+                }
+                if (((UnicodeString) key).getString().equals("authData")) {
+                    byte[] authData = ((ByteString) attObjMap.get(key)).getBytes();
+                    attestationObject = addAuthData(authData, attestationObject);
```

Comment:
this auth data is just for the 'none' attestation case. I think we need to make a choice on the 'fmt' value to choose other formats.